frontend/src/component/insights/impact-metrics/ImpactMetrics.tsx

```diff
@@ -0,0 +1,294 @@
+                timeSeriesData.length === 0 ||
+                !data.datasets.some((d) => d.data.length > 1)),
+        [data, isLoading, timeSeriesData],
+    );
+
+    const minTime = start
+        ? fromUnixTime(Number.parseInt(start, 10))
+        : undefined;
+    const maxTime = end ? fromUnixTime(Number.parseInt(end, 10)) : undefined;
+
+    const placeholder = selectedSeries ? (
+        <NotEnoughData description='Send impact metrics using Unleash SDK and select data series to view the chart.' />
+    ) : (
+        <NotEnoughData
+            title='Select a metric series to view the chart.'
+            description=''
+        />
+    );
+    const cover = notEnoughData ? placeholder : isLoading;
+
+    return (
+        <InsightsSection title='Impact metrics'>
+            <StyledWidget>
+                <StyledWidgetStats>
+                    <Box
+                        sx={(theme) => ({
+                            display: 'flex',
+                            flexDirection: 'column',
+                            gap: theme.spacing(2),
+                            width: '100%',
+                        })}
+                    >
+                        <ImpactMetricsControls
+                            selectedSeries={selectedSeries}
+                            onSeriesChange={handleSeriesChange}
+                            selectedRange={selectedRange}
+                            onRangeChange={setSelectedRange}
+                            beginAtZero={beginAtZero}
+                            onBeginAtZeroChange={setBeginAtZero}
+                            metricSeries={metricSeries}
+                            loading={metadataLoading}
+                            selectedLabels={selectedLabels}
+                            onLabelsChange={setSelectedLabels}
+                            availableLabels={availableLabels}
+                        />
+
+                        {!selectedSeries && !isLoading ? (
+                            <Typography variant='body2' color='text.secondary'>
+                                Select a metric series to view the chart
+                            </Typography>
+                        ) : null}
+                    </Box>
+                </StyledWidgetStats>
+
+                <StyledChartContainer>
+                    {hasError ? (
+                        <Alert severity='error'>
+                            Failed to load impact metrics. Please check if
+                            Prometheus is configured and the feature flag is
+                            enabled.
+                        </Alert>
+                    ) : null}
+                    <LineChart
+                        data={
+                            notEnoughData || isLoading ? placeholderData : data
+                        }
+                        overrideOptions={
+                            shouldShowPlaceholder
+                                ? {}
+                                : {
```

Comment:
there's a lot of of whitespace on the left hear = code complexity (https://codescene.io/docs/guides/technical/complexity-trends.html#what-s-complexity-anyway)